src/Core/OrganizationFeatures/OrganizationUsers/AcceptOrgUserCommand.cs

```diff
@@ -0,0 +1,224 @@
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.OrganizationFeatures.OrganizationUsers.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationUsers;
+
+public class AcceptOrgUserCommand : IAcceptOrgUserCommand
+{
+    private readonly IDataProtector _dataProtector;
+    private readonly IGlobalSettings _globalSettings;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IPolicyService _policyService;
+    private readonly IMailService _mailService;
+    private readonly IUserRepository _userRepository;
+    private readonly IDataProtectorTokenFactory<OrgUserInviteTokenable> _orgUserInviteTokenDataFactory;
+
+    public AcceptOrgUserCommand(
+        IDataProtectionProvider dataProtectionProvider,
+        IGlobalSettings globalSettings,
+        IOrganizationUserRepository organizationUserRepository,
+        IOrganizationRepository organizationRepository,
+        IPolicyService policyService,
+        IMailService mailService,
+        IUserRepository userRepository,
+        IDataProtectorTokenFactory<OrgUserInviteTokenable> orgUserInviteTokenDataFactory)
+    {
+
+        // TODO: remove data protector when old token validation removed
+        _dataProtector = dataProtectionProvider.CreateProtector("OrganizationServiceDataProtector");
+        _globalSettings = globalSettings;
+        _organizationUserRepository = organizationUserRepository;
+        _organizationRepository = organizationRepository;
+        _policyService = policyService;
+        _mailService = mailService;
+        _userRepository = userRepository;
+        _orgUserInviteTokenDataFactory = orgUserInviteTokenDataFactory;
+    }
+
+    public async Task<OrganizationUser> AcceptOrgUserByEmailTokenAsync(Guid organizationUserId, User user, string emailToken,
+        IUserService userService)
+    {
+        var orgUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
+        if (orgUser == null)
+        {
+            throw new BadRequestException("User invalid.");
+        }
+
+        // Tokens will have been created in two ways in the OrganizationService invite methods:
+        // 1. New way - via OrgUserInviteTokenable
+        // 2. Old way - via manual process using data protector initialized with purpose: "OrganizationServiceDataProtector"
+        // For backwards compatibility, must check validity of both types of tokens and accept if either is valid
+
+        // TODO: PM-4142 - remove old token validation logic once 3 releases of backwards compatibility are complete
+        // TODO: update this code to only run old token validation if new token validation fails
```

Comment:
suggestion: Looks like this second TODO is already implemented below and the comment can be removed.